tests/test_matchbox.py

```diff
@@ -0,0 +1,123 @@
+"""Tests for MatchBox."""
+from matchbox.matchbox import MatchBox
+
+
+class IndexedObject(object):
+
+    """Dummy object to be indexed."""
+
+    def __init__(self, value, matches=True):
+        """
+        Dummy object's initializing method.
+
+        :param str value: value for a characteristic.
+        :param bool matches: whether it matches or not.
+        """
+        self.characteristic = value
+        self.characteristic_match = matches
+
+
+def test_matchbox_indexed_empty_characteristic():
+    """Check simple adding object to index if it does match characteristic's value."""
+    ob = IndexedObject(None)
+
+    matchbox = MatchBox('characteristic')
+    matchbox.add(ob)
+    assert not matchbox.index, "index should be empty."
+    assert not matchbox.exclude_unknown, "collection for not matching unknown should also be empty."
+
+
+def test_matchbox_indexed_match():
+    """Check simple adding object to index if it does match characteristic's value."""
+    ob = IndexedObject('x')
+
+    matchbox = MatchBox('characteristic')
+    matchbox.add(ob)
+    assert ob.characteristic in matchbox.index, "characteristic value should result in an entry in index."
+    assert matchbox.index[ob.characteristic] == set(), "But the entry should be empty."
+    assert ob in matchbox.exclude_unknown, "object should be not matching any future entries though."
+
+
+def test_matchbox_indexed_not_match():
+    """Check simple adding object to index if it does not match a certain characteristic's value."""
+    ob = IndexedObject('x', False)
+
+    matchbox = MatchBox('characteristic')
+    matchbox.add(ob)
+    assert ob.characteristic in matchbox.index, "characteristic value should result in an entry in index."
+    assert ob in matchbox.index[ob.characteristic], "Object should be in set under it's characteristic's value key."
+    assert matchbox.exclude_unknown == set(), "object should not be matching any unknown values."
+
+
+def test_matchbox_mixed_objects():
+    """
+    Test more complex example.
+
+    +--------+-------+-------------------------+
```

Comment:
Yeah, wanted to point out what it will be at the same time, if ever the example in add will change, test will adhere to closely placed example.